contracts/HATVault.sol

```diff
@@ -527,18 +520,16 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
     /** @notice See {IHATVault-withdraw}. */
     function withdraw(uint256 assets, address receiver, address owner) 
         public override(IHATVault, ERC4626Upgradeable) virtual returns (uint256) {
-
         uint256 shares = previewWithdraw(assets);
-        uint256 fee = _convertToAssets(shares - _convertToShares(assets, MathUpgradeable.Rounding.Up), MathUpgradeable.Rounding.Up);
+        uint256 fee = assets * withdrawalFee / (HUNDRED_PERCENT - withdrawalFee);
```

Comment:
Why is the fee calculation happening here and then passed to _withdraw? Similar logic is duplicated in `redeem`. It all gets confusing bc previewWithdraw takes into account the fees as well.

Perhaps create a `_withdraw(caller, reciever, shares)` returns (assets, fee)` function that handles the calculation in one place?

Reply:
Ok I made a function to return both, so fee calculations happen in a single place